### PR TITLE
ft: S3C 1171 list objects v2

### PR DESCRIPTION
**Pull request by dora-korpar**
This implements v2 of the GET Bucket API (https://docs.aws.amazon.com/AmazonS3/latest/API/v2-RESTBucketGET.html)

There is also a design doc here that highlights the differences between v1 & v2
https://github.com/scality/cloudserver/commit/9372f7ae1dc99f0252943b4fdac468cfd9f3b7fb

### CODE REVIEW

**Review by alexanderchan-scality — APPROVED** [review]
lgtm

File: lib/algos/list/delimiterMaster.js [hunk 1]
```diff
@@ -57,9 +60,10 @@ class DelimiterMaster extends Delimiter {
             key = key.slice(0, versionIdIndex);
             /* - key === this.prvKey is triggered when a master version has
              *   been accepted for this key,
-             * - key === this.NextMarker is triggered when a listing page ends
-             *   on a accepted obj and the next page starts with a version of
-             *   this object. In that case prvKey is default set to undefined
+             * - key === this.NextMarker or this.NextContinueToken is triggered
+             *   when a listing page ends on an accepted obj and the next page
+             *   starts with a version of this object.
+             *   In that case prvKey is default set to undefined
```
Comment (bennettbuchanan) [inline lib/algos/list/delimiterMaster.js:66]: ~is default~ is by default

File: tests/unit/algos/list/delimiter.js [hunk 2]
```diff
@@ -273,6 +273,181 @@ const tests = [
...
+            receivedData[5],
+            receivedData[6],
+            receivedData[7],
+            receivedData[8],
+            receivedData[9],
+        ],
+        CommonPrefixes: [],
+        Delimiter: undefined,
+        IsTruncated: false,
+        NextContinuationToken: undefined,
```
Comment (bennettbuchanan) [inline tests/unit/algos/list/delimiter.js:300]: Thinking out loud here: are we interested in returning the `StartAfter` value, or is that just handled by Cloudserver in the XML response?

**Comment by bennettbuchanan** [comment]
@bert-e wait

**Review by bennettbuchanan — APPROVED** [review]
Left some minor comments. Overall, LGTM! 🎸 

I've added a bert-e wait in case we need to time it with CloudServer changes, so feel free to remove that when this is ready.

File: tests/unit/algos/list/delimiter.js (hunk 2 above)
Comment (dora-korpar) [reply tests/unit/algos/list/delimiter.js:300]: Yeah, it's handled by Cloudserver

File: lib/algos/list/delimiterMaster.js (hunk 1 above)
Comment (jeremyds) [reply lib/algos/list/delimiterMaster.js:66]: if you modify my code comment you have to fix my english @dora-korpar :rofl:

**Review by jeremyds — APPROVED** [review]

**Pull request merged**